packages/agoric-cli/src/lib/bundles.js

```diff
@@ -0,0 +1,88 @@
+/* global Buffer */
+import { BlobReader, TextWriter, ZipReader } from '@zip.js/zip.js';
+import fs from 'node:fs';
+import path from 'node:path';
+
+/** @import {Bundle} from '@agoric/swingset-vat'; */
+
+/**
+ * @typedef CompartmentMap
+ * @property {string[]} tags
+ * @property {{compartment: string, module: string}} entry
+ * @property {Record<string, {name: string, label: string, location: string, modules: Record<string, {compartment: string, module: string}>}} compartments
+ */
+
+/** @param {Bundle} bundleObj*/
+export const extractBundleInfo = async bundleObj => {
+  await null;
+  if (bundleObj.moduleFormat === 'endoZipBase64') {
+    const contents = Buffer.from(bundleObj.endoZipBase64, 'base64');
+    const zipBlob = new Blob([contents], { type: 'application/zip' });
+
+    const zipFileReader = new BlobReader(zipBlob);
+    const writer = new TextWriter('utf-8');
+
+    const zipReader = new ZipReader(zipFileReader);
+    const entries = await zipReader.getEntries();
+    await zipReader.close();
```

Comment:
Consider `@endo/zip` for the subset of zip we use and an API that flushes with compartment mapper. But, the above will do, just duplicate a dependency.

Reply:
I tried but there was no obvious way to get `compressedSize` so I left as is (with a comment)

Reply:
Neither bundleSource nor importBundle supports compression. These zips are legible in a hex dump.

Reply:
Ah, that makes things simpler